src/OmniSharp/Api/Highlighting/OmnisharpController.Highlighting.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Models;
+
+namespace OmniSharp
+{
+    public partial class OmnisharpController
+    {
+        [HttpPost("highlight")]
+        public async Task<IEnumerable<HighlightResponse>> Highlight(HighlightRequest request)
+        {
+            var document = _workspace.GetDocument(request.FileName);
```

Comment:
@Alxandr With DNX you can define different projects (check project.json), basically for core_clr and desktop_clr. Internally, Omnisharp maps that to different Roslyn projects. Therefore you can have multiple documents for the same filepath. So depending on the containing project of a document a line of code might be actually doing something or not. VS15 has a nice representation for this:

![image](https://cloud.githubusercontent.com/assets/1794099/7429179/3769e7da-effd-11e4-9fee-b62115b3bad2.png)